### PR TITLE
create task to identify and delete orphaned CloudAccount objects

### PR DESCRIPTION
**Pull request by infinitewarp**
https://github.com/cloudigrade/cloudigrade/issues/1070

### CODE REVIEW

File: cloudigrade/api/tests/tasks/maintenance/test_delete_orphaned_cloud_accounts.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+            old_healthy_accounts
+            + old_orphaned_accounts
+            + new_healthy_accounts
+            + new_orphaned_accounts
+        )
+        expected_cloud_accounts_after = (
+            old_healthy_accounts + new_healthy_accounts + new_orphaned_accounts
+        )
+
+        self.assertTrue(
```
Comment (infinitewarp) [inline cloudigrade/api/tests/tasks/maintenance/test_delete_orphaned_cloud_accounts.py:80]: whoops. should be `assertEqual` Fixing this locally.

File: cloudigrade/api/tests/tasks/maintenance/test_delete_orphaned_cloud_accounts.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        long_ago = util_helper.utc_dt(2018, 1, 5, 0, 0, 0)
+        recently = util_helper.utc_dt(2021, 11, 17, 0, 0, 0)
+
+        with util_helper.clouditardis(long_ago):
+            # We need to generate in the past so they are old enough to be found.
+            healthy_accounts, orphaned_accounts = self.generate_accounts()
+
+        expected_cloud_accounts = healthy_accounts + orphaned_accounts
+
+        self.assertTrue(
```
Comment (infinitewarp) [inline cloudigrade/api/tests/tasks/maintenance/test_delete_orphaned_cloud_accounts.py:140]: ditto above. should be `assertEqual`.

File: cloudigrade/api/tests/tasks/maintenance/test_delete_orphaned_cloud_accounts.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+            for account in orphaned_accounts
+        }
+        error_messages = {
+            record.message
+            for record in logging_watcher.records
+            if record.levelname == "ERROR"
+        }
+        self.assertEqual(expected_error_messages, error_messages)
+
+        self.assertTrue(
```
Comment (infinitewarp) [inline cloudigrade/api/tests/tasks/maintenance/test_delete_orphaned_cloud_accounts.py:171]: ditto `assertEqual`

File: cloudigrade/api/tests/tasks/maintenance/test_delete_orphaned_cloud_accounts.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+        long_ago = util_helper.utc_dt(2018, 1, 5, 0, 0, 0)
+        recently = util_helper.utc_dt(2021, 11, 17, 0, 0, 0)
+
+        with util_helper.clouditardis(long_ago):
+            # We need to generate in the past so they are old enough to be found.
+            healthy_accounts, orphaned_accounts = self.generate_accounts()
+
+        expected_cloud_accounts = healthy_accounts + orphaned_accounts
+
+        self.assertTrue(
```
Comment (infinitewarp) [inline cloudigrade/api/tests/tasks/maintenance/test_delete_orphaned_cloud_accounts.py:188]: ditto `assertEqual` sheeeesh

File: cloudigrade/api/tests/tasks/maintenance/test_delete_orphaned_cloud_accounts.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+            for account in orphaned_accounts
+        }
+        warning_messages = {
+            record.message
+            for record in logging_watcher.records
+            if record.levelname == "WARNING"
+        }
+        self.assertEqual(expected_warning_messages, warning_messages)
+
+        self.assertTrue(
```
Comment (infinitewarp) [inline cloudigrade/api/tests/tasks/maintenance/test_delete_orphaned_cloud_accounts.py:226]: `assertEqual`

File: cloudigrade/api/tasks/maintenance.py [hunk 6]
```diff
@@ -308,3 +311,106 @@ def _delete_cloud_account_related_objects(cloud_account):
...
+    found_orphans = []
+    deleted_orphans = []
+    for cloud_account in CloudAccount.objects.filter(updated_at__lt=max_updated_at):
+        if not cloud_account.content_object:
+            found_orphans.append(cloud_account)
+            logger.info(
+                _("Found orphan %(cloud_account)s"), {"cloud_account": cloud_account}
+            )
+            try:
+                if source := sources.get_source():
```
Comment (abellotti) [inline cloudigrade/api/tasks/maintenance.py:354]: don't we need to pass the account number ?

Comment (infinitewarp) [reply cloudigrade/api/tasks/maintenance.py]: Indeed we do! I also spotted this locally but completely forgot to push it. Fixed now. 😅

**Review by katherine-black — APPROVED** [review]

**Review by katherine-black — APPROVED** [review]

**Pull request merged**